File: VB/WpfReportDesigner_local/LocalReportServiceClient.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraReports.Service
Imports DevExpress.Data.Utils.ServiceModel
Imports System.ComponentModel

Namespace WpfReportDesigner_local
	Friend Class LocalReportServiceClient
		Private ReadOnly reportService_Renamed As ReportService
		Protected ReadOnly Property ReportService() As ReportService
			Get
				Return reportService_Renamed
			End Get
		End Property

		Public Sub New()
			reportService_Renamed = New LocalReportService()
		End Sub

		Protected Sub RaiseScalarOperationCompletedEvent(Of T)(ByVal completedEvent As EventHandler(Of ScalarOperationCompletedEventArgs(Of T)), ByVal result As Object, ByVal asyncState As Object)
			If completedEvent IsNot Nothing Then
				completedEvent(Me, New ScalarOperationCompletedEventArgs(Of T)(result, Nothing, False, asyncState))
			End If
		End Sub

		Protected Sub RaiseOperationCompletedEvent(ByVal completedEvent As EventHandler(Of AsyncCompletedEventArgs), ByVal asyncState As Object)
			If completedEvent IsNot Nothing Then
				completedEvent(Me, New AsyncCompletedEventArgs(Nothing, False, asyncState))
			End If
		End Sub
	End Class
End Namespace

File: CS/WpfReportDesigner_local/LocalReportServiceClient.cs
using System;
using DevExpress.XtraReports.Service;
using DevExpress.Data.Utils.ServiceModel;
using System.ComponentModel;

namespace WpfReportDesigner_local {
    class LocalReportServiceClient {
        readonly ReportService reportService;
        protected ReportService ReportService { get { return reportService; } }

        public LocalReportServiceClient() {
            reportService = new LocalReportService();
        }

        protected void RaiseScalarOperationCompletedEvent<T>(EventHandler<ScalarOperationCompletedEventArgs<T>> completedEvent, object result, object asyncState) {
            if(completedEvent != null)
                completedEvent(this, new ScalarOperationCompletedEventArgs<T>(result, null, false, asyncState));
        }

        protected void RaiseOperationCompletedEvent(EventHandler<AsyncCompletedEventArgs> completedEvent, object asyncState) {
            if(completedEvent != null)
                completedEvent(this, new AsyncCompletedEventArgs(null, false, asyncState));
        }
    }
}
